src/segments/argocd.go

```diff
@@ -0,0 +1,130 @@
+package segments
+
+import (
+	"errors"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v3"
+	"os"
+	"path"
+	"strings"
+)
+
+const (
+	argocdOptsEnv     = "ARGOCD_OPTS"
+	argocdInvalidFlag = "invalid flag"
+	argocdInvalidYaml = "invalid yaml"
+	argocdNoCurrent   = "no current context"
+)
+
+type ArgocdContext struct {
+	Name   string `yaml:"name"`
+	Server string `yaml:"server"`
+	User   string `yaml:"user"`
+}
+
+type ArgocdConfig struct {
+	Contexts       []*ArgocdContext `yaml:"contexts"`
+	CurrentContext string           `yaml:"current-context"`
+}
+
+type Argocd struct {
+	props properties.Properties
+	env   platform.Environment
+
+	Context ArgocdContext
+	flags   *pflag.FlagSet
+}
+
+func (a *Argocd) Template() string {
+	return " {{ .Context.Name }} "
+}
+
+func (a *Argocd) Init(props properties.Properties, env platform.Environment) {
+	a.props = props
+	a.env = env
+}
+
+func (a *Argocd) Enabled() bool {
+	// always parse config instead of using cli to save time
+	configPath := a.getConfigPath()
+	succeeded, err := a.parseConfig(configPath)
+	if err != nil {
+		a.env.Error(err)
+		return false
+	}
+	return succeeded
+}
+
+func (a *Argocd) getConfigPath() string {
+	cp := path.Join(a.env.Home(), ".config", "argocd", "config")
+	cpo, err := a.getConfigFromOpts()
+	if err != nil {
+		a.env.Error(err)
+		return cp
+	}
+	if len(cpo) > 0 {
+		cp = cpo
+	}
+	return cp
+}
+
+func (a *Argocd) getConfigFromOpts() (string, error) {
+	// don't exit/panic when encountering invalid flags
+	a.flags = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
```

Comment:
why do we use all flags here? Can't we only add the config flag and get that value? I would also not make that part of the struct, but an inline variable as it's only used here.

Reply:
because i was trying to tell the difference between valid argocd flags and invalid ones. so that we can handle the use case like [this](https://github.com/JanDeDobbeleer/oh-my-posh/pull/3684/files#diff-b58057c229fe7937c24b999999dad7b3bdefbb75e3513b6d0922d1b1b9eaf412R24). if those valid flags got removed, pflags would fail to retrieve the config flag we need.

Reply:
updated and removed the error.

Reply:
> an inline variable as it's only used here

agreed and updated.